docs/comparison-matrix.md

```diff
@@ -0,0 +1,42 @@
+<!-- markdownlint-disable-file -->
+
+# Comparison Matrix 
+
+Legend:
+
+- 🟢 Available (grading added in some cases spanning: Bad, Average, Good, Excellent)
+- 🟡 In development
+- 🔵 Potential future feature
+- 🔴 Not currently available
+- ❔ Not clear
+
+| **Feature**                                            | **Proto Buffers** | **ADL**      | **JSON Schema** | **Lambda Buffers** | **CDDL** | **ASN.1**    |
+|--------------------------------------------------------|-------------------|--------------|-----------------|--------------------|----------|--------------|
+| Sum Types                                              | 🟢                | 🟢           | 🔴              | 🟢                 | 🟢       | 🟢           |
+| Record Types                                           | 🟢                | 🟢           | 🟢              | 🟢                 | 🟢       | 🟢           |
+| Product Types                                          | 🔴                | 🔴           | 🔴              | 🟢                 | ❔       | 🔴           |
+| Recursive Types                                        | 🟢                | 🟢           | 🔴              | 🟢                 | 🟢       | ❔           |
+| Type functions (Generics)                              | 🔴                | 🟢           | 🔴              | 🟢                 | 🟢       | 🔴           |
```

Comment:
Maybe we should call this "Parameterized types" or something like that? 

"Type functions" is accurate (esp from our point of view because that's how things get kind-checked) but I think "Parameterized types" is what most protobufs users would call that feature (which, based on some cursory research, seems to be one of the most-desired features in a protobuf competitor)

Reply:
Makes sense!

Reply:
Done!